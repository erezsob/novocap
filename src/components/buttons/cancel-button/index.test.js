import React from 'react';
import { shallow } from 'enzyme';
import Button from 'material-ui/Button';
import renderer from 'react-test-renderer';
import CancelButton from './';

describe('<CancelButton />', () => {
  const mockAction = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CancelButton action={mockAction} />);
  });

  it('renders', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('renders properly', () => {
    const tree = renderer.create(<CancelButton action={mockAction} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('triggers an action onClick event', () => {
    wrapper
      .dive()
      .find(Button)
      .simulate('click');
    expect(mockAction).toHaveBeenCalledTimes(1);
  });
});
