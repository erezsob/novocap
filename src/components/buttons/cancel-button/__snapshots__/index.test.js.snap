// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelButton /> renders properly 1`] = `
<button
  className="MuiButtonBase-root-17 MuiButton-root-2 CancelButton-cancel-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    Cancel
  </span>
  <span
    className="MuiTouchRipple-root-20"
  />
</button>
`;
